sim/core/spell_cast.go

```diff
@@ -135,31 +138,38 @@ func (spellEffect *SpellEffect) calculateDirectDamage(sim *Simulation, spellCast
 		damage *= spellCast.CritMultiplier
 	}
 
-	damage = calculateResists(sim, damage, spellEffect)
+	if !spellCast.Binary {
+		damage = calculateResists(sim, damage, spellEffect)
+	}
 
 	spellEffect.Damage = damage
 }
 
 func (spellEffect *SpellEffect) applyDot(sim *Simulation, spellCast *SpellCast, ddInput *DotDamageInput) {
 	totalSpellPower := spellCast.Character.GetStat(stats.SpellPower) + spellCast.Character.GetStat(spellCast.SpellSchool) + spellEffect.BonusSpellPower
 	// snapshot total damage per tick
-	ddInput.damagePerTick = (ddInput.TickBaseDamage + totalSpellPower*ddInput.TickSpellCoefficient) * spellEffect.DamageMultiplier
+	ddInput.damagePerTick = (ddInput.TickBaseDamage + totalSpellPower*ddInput.TickSpellCoefficient)
 	ddInput.finalTickTime = sim.CurrentTime + time.Duration(ddInput.NumberOfTicks)*ddInput.TickLength
 
 	pa := &PendingAction{
+		Name:         spellCast.Name,
 		NextActionAt: sim.CurrentTime + ddInput.TickLength,
 	}
 
 	pa.OnAction = func(sim *Simulation) {
-		damage := ddInput.damagePerTick
-		damage = calculateResists(sim, damage, spellEffect)
+		damage := spellCast.Character.OnPeriodicDamage(sim, spellCast, spellEffect, ddInput.damagePerTick)
+		damage = spellEffect.Target.OnPeriodicDamage(sim, spellCast, spellEffect, damage)
+
+		if !spellCast.Binary {
+			damage = calculateResists(sim, damage, spellEffect)
```

Comment:
added check for binary spells to skip resist checks (also in direct damage checks above)